### PR TITLE
Frames for the end effector module

### PR DESCRIPTION
**Pull request by VictorAdamse01**

### CODE REVIEW

File: code/headers/frame_types.hpp [hunk 1]
```diff
@@ -594,6 +596,31 @@ namespace r2d2 {
...
+    /**
+     * This frame is used to request the type of the end effector
+     * 
+     * End effector wiki:
+     * https://github.com/R2D2-2019/R2D2-2019/wiki/End-Effectors#2-Interface
+     */
+    R2D2_PACK_STRUCT
+    struct frame_end_effector_type_s {
+        //the type of end effector 
+        enum end_effector_type { CLAW, NONE };
```
Comment (itzandroidtab) [inline code/headers/frame_types.hpp]: This enum should be moved to the enum file

**Review by itzandroidtab — CHANGES_REQUESTED** [review]

**Comment by itzandroidtab** [comment]
Can't these frames be combined? And when you do a request it will respond with the end effector type with its status.

**Comment by VictorAdamse01** [comment]
The frames can be combined but this might give issues when we start adding multiple end effectors that require different frames with multiple values.

File: code/headers/frame_types.hpp [hunk 2]
```diff
@@ -594,6 +596,30 @@ namespace r2d2 {
...
     };
 
+    /**
+     * This frame is used to request the type of the end effector
+     * 
+     * End effector wiki:
+     * https://github.com/R2D2-2019/R2D2-2019/wiki/End-Effectors#2-Interface
+     */
+    R2D2_PACK_STRUCT
+    struct frame_end_effector_type_s {
```
Comment (LRstudentHU) [inline code/headers/frame_types.hpp:682]: Why do we require this struct? Each end effector has it's own frame after all

**Review by LRstudentHU — CHANGES_REQUESTED** [review]

File: code/headers/frame_types.hpp (hunk 2 above)
Comment (VictorAdamse01) [reply code/headers/frame_types.hpp:682]: so the other modules can see what end effector is attached

**Review by LRstudentHU — APPROVED** [review]

**Review by itzandroidtab — APPROVED** [review]

**Pull request merged**